src/d/actor/d_a_wbird.cpp

```diff
@@ -5,70 +5,208 @@
+    if(field_0x29D){
+        field_0x29D = false;
+        mDoAud_seStart(JA_SE_TAKT_WIND_DEMO);
+    }
+    if (field_0x29E > 0) {
+        field_0x29E--;
+        if (field_0x29E > 60) {
+            fopAcM_posMove(this, NULL);
+            speed.y = speed.y + field_0x2A0;
+        }
+        if (field_0x29E < 15) {
+            dKyw_custom_windpower(0.0f);
+        }
+        if (field_0x29E == 70) {
+            dKyw_tact_wind_set_go();
+        }
+    } else {
+        dComIfGp_evmng_cutEnd(dComIfGp_evmng_getMyStaffId("WINDMAN"));
+        player->changeDemoMoveAngle(mAngle);
+        mAction = 1;
+        mDoAud_seStart(JA_SE_TAKT_WIND_END);
+    }
 }
 
 /* 000005DC-00000850       .text actionSelect__9daWbird_cFv */
 void daWbird_c::actionSelect() {
-    /* Nonmatching */
+    short sVar2 = field_0x29E;
+    if (sVar2 == 10) {
+        dComIfGp_setOperateWindOn();
+        mDoAud_seStart(JA_SE_TAKT_WIND_DISP);
+        field_0x29E++;
+    } else if(sVar2 > 10){
+        mDoAud_seStart(JA_SE_SYS_WTAKT_WIND_AMB);
+        switch (dComIfGp_getOperateWind()) {
+            case 1: {
+                mDoAud_seStart(JA_SE_TAKT_WIND_DECIDE);
+                setStartPos();
+                mAction = 2;
+                daPy_py_c* player = daPy_getPlayerActorClass();
+                sVar2 = current.angle.y + 0x7fff;
+                mAngle = player->shape_angle.y;
+                player->setPlayerPosAndAngle(&player->current.pos, sVar2);
+                if(dComIfGp_checkPlayerStatus0(0, 0x00010000)){
+                    mEventIdx = dComIfGp_evmng_getEventIdx("TACT_WINDOW2_SHIP");
+                } else{
+                    mEventIdx = dComIfGp_evmng_getEventIdx("TACT_WINDOW2");
+                }
+                fopAcM_orderChangeEventId(this, mEventIdx, 0, 0xFFFF);
+                player = daPy_getPlayerActorClass();
+                player->cancelOriginalDemo();
+                dKyw_custom_windpower(windpower);
+                field_0x29D = true;
+                break;
+            }
+            case 0 : {
+                mDoAud_seStart(JA_SE_TAKT_WIND_CANCEL);
+                dComIfGp_event_reset();
+                fopAcM_delete(this);
+                break;
+            }
+        }
+    }
+    else {
+        field_0x29E++;
+    }
 }
-
 /* 00000850-00000858       .text daWbird_Draw__FP9daWbird_c */
 static BOOL daWbird_Draw(daWbird_c*) {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 00000858-000008D0       .text daWbird_Execute__FP9daWbird_c */
-static BOOL daWbird_Execute(daWbird_c*) {
-    /* Nonmatching */
+static BOOL daWbird_Execute(daWbird_c* i_this) {
+    switch(i_this->mAction) {
+    case 3:
+        i_this->actionSelect();
+        break;
+    case 2:
+        i_this->actionMove();
+        break;
+    case 1:
+        i_this->actionEnd();
+        break;
+    default:
+        i_this->actionWait();
+        break;
+    }
+    i_this->calcMtx();
+    return TRUE;
 }
```

Comment:
The `execute` and `draw` methods of this actor are inlined, so for example this logic can be moved into `daWbird_c::execute` and the static would then just be:
```suggestion
static BOOL daWbird_Execute(daWbird_c* i_this) {
    return i_this->execute();
}
```

For some actors this matters for matching so it's good to be aware of. For this one it doesn't, but still might as well get the code closer to how the original devs wrote it.

Reply:
i cant find daWbird_c::execute or draw in the files of vscode, do i need to create these functions or am i not searching correctly

Reply:
Yeah you'll need to create them. My script put declarations of them in the header files so you don't need to add anything there, but I didn't create empty placeholders in the .cpp file like I did for the smaller inlines that go in the headers, such as `setAction`